### PR TITLE
[FW-113] ethernet implemetation 

### PR DESCRIPTION
**Pull request by oganigl**
I've done 3 classes: DatagramSocket, Socket and Server. the packets received from the communication are keep in a queue so you have to ask to the class to receive the packets. I'm open to differents ways of doing this

### CODE REVIEW

**Comment by oganigl** [comment]
Everything has been tested

File: src/services/communications/DatagramSocket.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import socket
+import threading
+import time
+from queue import Queue
+from typing import Optional
+MAX_SIZE_PACKET = 1024
+MAX_LISTEN_CONNECTIONS = 5
+TIMEOUT_TIME = 2.0
+class DatagramSocket:
```
Comment (jmaralo) [inline src/services/communications/DatagramSocket.py:9]:
```suggestion
from typing import Optional

MAX_SIZE_PACKET = 1024
MAX_LISTEN_CONNECTIONS = 5
TIMEOUT_TIME = 2.0

class DatagramSocket:
```

File: src/services/communications/DatagramSocket.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        self._sock.settimeout(TIMEOUT_TIME) 
+        self._running = True
+        self._recv_thread = threading.Thread(target=self._receive, daemon=True)
+        self._recv_thread.start()      
+    
+    def _receive(self): 
+        while self._running:
+            try:
+                data,address = self._sock.recvfrom(MAX_SIZE_PACKET)
+                if address == (self.remote_ip,self.remote_port): #check if the receiving message comes from the ip address correct
```
Comment (jmaralo) [inline src/services/communications/DatagramSocket.py:27]: I feel like this comment is redundant

**Review by jmaralo — DISMISSED** [review]
If you tested it and it worked, LGTM, just a few comments about style, big nitpicks.

Maybe wait before merging the PR to see if naggle will be a big issue or not, if it can be disabled programatically it's better, but otherwise one of the steps of the test script will be to turn off and back on once finished the naggle algorithm

**Review by jmaralo — APPROVED** [review]

**Review by jdmarmen — APPROVED** [review]

**Pull request merged**